.gitignore

```diff
@@ -1,21 +1,42 @@
-.nyc_output
-coverage
-dist
-lerna-debug.log
+# Package manager
 node_modules
-npm-debug.log
-yarn-error.log
+
+# Build
+dist
 _webpackOutput
 
 # Generated metadata (only the root one is generated!)
-
 /metadata.toml
 
+# Cache
+.cache
+.eslintcache
+
+# Testing
+.nyc_output
+coverage
 
-# Phenomic things
+# Env
+.env.local
+.env.*.local
 
-*.bundle.js
+# Logging
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+lerna-debug.log*
+.pnpm-debug.log*
 
-# Cache
+# Editor
+.idea
+.vscode
+.history
+*.suo
```

Comment:
I have never seen these extensions:
* `.ntvs*`
* `.njsproj`
* `.sln`
* `.sw?` 

What are they?

Reply:
They are mostly related to Visual Studio. The two first are related to their Node.js tools. And the `.sln` is (like `*.suo`) related to [organising projects](https://docs.microsoft.com/en-us/visualstudio/extensibility/internals/solution-dot-sln-file) there.

`.sw?` takes care of vim temporary files